src/hist.jl

```diff
@@ -40,46 +40,54 @@ function histrange(v::AbstractArray{T}, n::Integer, closed::Symbol=:left) where
 end
 
 function histrange(lo::F, hi::F, n::Integer, closed::Symbol=:left) where F
-    if hi == lo
-        start = F(hi)
-        step = one(F)
+    closed == :left || closed ==:right || 
+        throw(ArgumentError("closed must be :left or :right"))
+    lo = float(lo)
+    hi = float(hi)
+    if abs(hi-lo) < n*10eps(F)
+        if closed == :left
+            hi = hi + n*10eps(F)
+        elseif closed ==:right
+            lo = lo - n*10eps(F)
+        else
+            # never happens
```

Comment:
I'd just delete the `else` block. The fact that this doesn't happen is apparent from the error check above.